internal/observability/event/context.go

```diff
@@ -267,28 +266,61 @@ func WriteSysEvent(ctx context.Context, caller Op, data map[string]interface{})
 		}
 	}
 
-	eventer := SysEventer()
-	if eventer == nil {
-		logger := hclog.New(nil)
-		logger.Error(fmt.Sprintf("%s: no eventer available to write sysevent: (%s) %+v", op, caller, data))
-		return
+	eventer, ok := EventerFromContext(ctx)
+	if !ok {
+		eventer = SysEventer()
+		if eventer == nil {
+			logger := hclog.New(nil)
+			logger.Error(fmt.Sprintf("%s: no eventer available to write sysevent: (%s) %+v", op, caller, info))
+			return
+		}
 	}
 
 	id, err := newId(string(SystemType))
 	if err != nil {
 		eventer.logger.Error(fmt.Sprintf("%s: %v", op, err))
-		eventer.logger.Error(fmt.Sprintf("%s: unable to generate id while writing sysevent: (%s) %+v", op, caller, data))
+		eventer.logger.Error(fmt.Sprintf("%s: unable to generate id while writing sysevent: (%s) %+v", op, caller, info))
 	}
 
 	e := &sysEvent{
 		Id:      Id(id),
 		Version: sysVersion,
 		Op:      caller,
-		Data:    data,
+		Data:    info,
 	}
 	if err := eventer.writeSysEvent(ctx, e); err != nil {
 		eventer.logger.Error(fmt.Sprintf("%s: %v", op, err))
 		eventer.logger.Error(fmt.Sprintf("%s: unable to write sysevent: (%s) %+v", op, caller, e))
 		return
 	}
 }
+
+// MissingKey defines a key to be used as the "missing key" when ConvertArgs has
+// an odd number of args (it's missing a key in its key/value pairs)
+const MissingKey = "EXTRA_VALUE_AT_END"
+
+// ConvertArgs will convert the key/value pair args to a map.  If the args
+// provided are an odd number (they're missing a key in their key/value pairs)
+// then MissingKey is used to the missing key.
+func ConvertArgs(args ...interface{}) map[string]interface{} {
+	if len(args) == 0 {
+		return nil
+	}
+	if len(args)%2 != 0 {
+		extra := args[len(args)-1]
+		args = append(args[:len(args)-1], MissingKey, extra)
+	}
+
+	m := map[string]interface{}{}
+	for i := 0; i < len(args); i = i + 2 {
+		var key string
+		switch st := args[i].(type) {
+		case string:
+			key = st
+		default:
```

Comment:
Curious if you think this is what we want, as opposed to forcing it to be a string. OTOH I guess you're trying to avoid having to return an error or panic...

Reply:
I do think this is a reasonable default and eliminates the need for an error or panic.